### PR TITLE
lagrangian: Added generic_potential_energy argument to lagrangian

### LINKED ISSUE #11967: There is no clean way to define a system's potential energy expression in LagrangesMethod
When forming Lagrange's equations the potential energy expression, in general, is composed of individual components that map to particles and bodies. For example, if a spring is connected between two particles, the potential energy expression will look like `k/2/x**2` where `x` is the distance between the two particles. This is a system wide definition of the spring's potential energy and can't necessarily be mapped to the `.potential_energy` attribute on the two particles. Maybe the `lagrangian` function needs an optional keyword argument that accepts generic potential energy expressions.

### PR DESCRIPTION
**Pull request by bhavishyagopesh**
Added generic_potential_energy to the potential energy which cannot be
mapped to the .potential_energy attribute of particles.

Fixes #11967

### CODE REVIEW

**Comment by bhavishyagopesh** [comment]
@moorepants I'm unable to spot the error,it says syntactical error in the new function but I'm unable to spot it. HELP.

**Comment by jksuom** [comment]
Named parameters should probably come before `*`-parameters.

**Comment by bhavishyagopesh** [comment]
@moorepants  One job suffered time out, can I tell whats wrong so that I could rebase and commit.

**Comment by bhavishyagopesh** [comment]
Ping @jksuom ,I just rebased the commits and its failing in some printing test, could you please fix this and merge this.

**Comment by czgdp1807** [comment]
ping @jksuom @bjodah Any comments on the current state of this PR?

File: sympy/physics/mechanics/functions.py [hunk 1]
```diff
@@ -365,6 +365,9 @@ def Lagrangian(frame, *body):
     body1, body2, body3... : Particle and/or RigidBody
         The body (or bodies) whose Lagrangian is required.
 
+    kwargs : generic_potential_energy
+        The potential energy which cannot be mapped to the .potential_energy
+        attribute of particles.
```
Comment (moorepants) [inline sympy/physics/mechanics/functions.py]: Please provide both an example where this is needed and a unit test for it.

**Comment by czgdp1807** [comment]
ping @bhavishyagopesh Thanks for the contribution. Would you still like to continue your work on this PR? Please address the latest review received.

**Comment by czgdp1807** [comment]
@moorepants Probably, this PR can also be extended. If you feel the same then please let me know. I will extend this and make a new PR to complete this. Since, I am unfamiliar with this module, it would be great if someone experienced picks this up. Please comment below if anyone is interested.

**Comment by czgdp1807** [comment]
@moorepants Your comments on, https://github.com/sympy/sympy/pull/12354#issuecomment-539350613

**Comment by moorepants** [comment]
I'm going to close this on. It is has many unrelated changes and there is not enough information to make any decision about this.

**Pull request closed without merging**